source/common/router/router.h

```diff
@@ -303,12 +321,21 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     void onBelowWriteBufferLowWatermark() override { enableDataFromDownstream(); }
 
     void disableDataFromDownstream() {
-      ASSERT(parent_.upstream_requests_.size() == 1);
+      // If there is only one upstream request, we can be assured that
+      // disabling reads will not slow down other upstream requests. If we've
+      // already seen the full downstream requst (downstream_end_stream_) then
+      // disabling reads is a no-op.
+      ASSERT(parent_.upstream_requests_.size() == 1 || downstream_end_stream_);
```

Comment:
This assert isn't always true until https://github.com/envoyproxy/envoy/pull/6643/files is merged because we might hit a per try timeout before `downstream_end_stream_` is true